Editor/foleys_GUITreeEditor.cpp

```diff
@@ -76,5 +109,58 @@ void GUITreeEditor::valueTreeParentChanged (juce::ValueTree& treeWhoseParentHasC
 {
 }
 
+GUITreeEditor::GuiTreeItem::GuiTreeItem (GUITreeEditor& refGuiTreeEditor, juce::ValueTree& refValueTree)
+    : guiTreeEditor {refGuiTreeEditor}, tree {refValueTree}
+{
+}
+
+juce::String GUITreeEditor::GuiTreeItem::getUniqueName() const
+{
+    return tree.getType().toString();
+}
+
+bool GUITreeEditor::GuiTreeItem::mightContainSubItems()
+{
+    return tree.getNumChildren() > 0;
+}
+
+void GUITreeEditor::GuiTreeItem::paintItem (juce::Graphics& g, int width, int height)
+{
+    if (isSelected())
+        g.fillAll ({91, 103, 109});
+
+    g.setColour (juce::Colours::white);
+    g.setFont (height * 0.7f);
+
+    juce::String s = tree.getType().toString();
+
+    juce::String ss = tree["caption"].toString();
```

Comment:
caption is not the property the user is interested in... it should be id, since caption is most times empty